dask/dataframe/shuffle.py

```diff
@@ -208,7 +211,7 @@ def set_partition(
             column_dtype=df.columns.dtype,
         )
 
-    df4.divisions = divisions
+    df4.divisions = [v for v in divisions.values]
```

Comment:
It may have been broken before, but this change may change the behavior of date time with tz.

```python
In [49]: s = pd.Series(pd.date_range('2000', periods=4, tz='US/Central'))

In [50]: s
Out[50]:
0   2000-01-01 00:00:00-06:00
1   2000-01-02 00:00:00-06:00
2   2000-01-03 00:00:00-06:00
3   2000-01-04 00:00:00-06:00
dtype: datetime64[ns, US/Central]

In [51]: s.values
Out[51]:
array(['2000-01-01T06:00:00.000000000', '2000-01-02T06:00:00.000000000',
       '2000-01-03T06:00:00.000000000', '2000-01-04T06:00:00.000000000'],
      dtype='datetime64[ns]')
```

In pandas, `Series[datetime64[ns, tz]].values` converts to UTC and drops the timezone.

Would the following code hit this? Can you check that it's unchanged?

```python
In [58]: df = pd.DataFrame({"A": pd.date_range('2000', periods=12, tz="US/Central"), "B": 1})

In [59]: ddf = dd.from_pandas(df, 2)

In [60]: ddf.set_index('A').divisions
Out[60]:
(Timestamp('2000-01-01 00:00:00-0600', tz='US/Central', freq='D'),
 Timestamp('2000-01-07 00:00:00-0600', tz='US/Central', freq='D'),
 Timestamp('2000-01-12 00:00:00-0600', tz='US/Central', freq='D'))
```

Reply:
Good eye!

I actually ran into this problem with one of the existing tests, and *I thought* that I fixed it by using `df4.divisions = [v for v in divisions.values]`.  Unfortunately, it looks like this line should be: `df4.divisions = [v for v in divisions]` , and even with this change, `freq` is getting dropped during the round trip.

The problem actually doesn't show up in the case you shared, but it does when you force the `shuffle` version of `set_index`.  For example:

```python
df = pd.DataFrame({"A": pd.date_range('2000', periods=12, tz="US/Central"), "B": 1})
ddf = dd.from_pandas(df, 2)
divisions = (
    pd.Timestamp('2000-01-01 00:00:00-0600', tz='US/Central', freq='D'),
    pd.Timestamp('2000-01-12 00:00:00-0600', tz='US/Central', freq='D')
)
ddf.set_index('A', divisions=divisions).divisions
```

Using `[v for v in divisions]` or `divisions.tolist()`, I get:

```
(Timestamp('2000-01-01 00:00:00-0600', tz='US/Central'),
 Timestamp('2000-01-12 00:00:00-0600', tz='US/Central'))
```

For a pandas-only workflow, the "better" solution is probably `divisions.tolist()`.  However, `tolist` is not yet available in cudf - So, I just went with the simple workaround.   I'm not quite sure how to address the missing `freq` property...

Reply:
> freq is getting dropped during the round trip.

That *may* not be worth worrying about. For date times, pandas mostly just uses `.freq` as an optimization.

```python
In [17]: a = pd.date_range('2000', periods=12, freq='S')[0]

In [18]: b = pd.date_range('2000', periods=12, freq='T')[0]

In [19]: a == b
Out[19]: True
```

So I think the only thing that matters is ensuring that the scalars in `.divisions` are datetimetz, not tz-naive. Can you verify that?

Reply:
Yes - I believe that [this test](https://github.com/dask/dask/blob/777bd66edf749fecd137c240e18f167fa1fc7813/dask/dataframe/tests/test_shuffle.py#L873) verifies that the tz is preserved in `divisions`.